src/test/scala/nodes/images/ConvolverSuite.scala

```diff
@@ -136,4 +136,73 @@ class ConvolverSuite extends FunSuite with LocalSparkContext with Logging {
+
+    val imArray = imageBGR.toArray
+
+    val imArrayRGB  = TestUtils.BGRtoRGB(new DenseMatrix(3, 256*256, imArray)).data
+
+     val image = new ChannelMajorArrayVectorizedImage(imArrayRGB, imageBGR.metadata)
+
+    val theanoFile = new File(TestUtils.getTestResourceFileName("images/theano_conv2d_nostride.csv"))
+
+    val convolvedImgRaw:Array[Double] = csvread(theanoFile).data
+
+    val convolvedImg = new ColumnMajorArrayVectorizedImage(convolvedImgRaw, ImageMetadata(imgWidth - convSize + 1, imgHeight - convSize + 1, numFilters))
+
+    val convBank = convert(new DenseMatrix(2, 5*5*3, (0 until 2*5*5*3).toArray.map(x => 1)), Double)
+
+    val convolver = new Convolver(convBank, imgWidth, imgHeight, imgChannels)
+
+    val poolImage = convolver(image)
+
+    logInfo(s"Image Dimensions ${poolImage.metadata.xDim} ${poolImage.metadata.yDim} ${poolImage.metadata.numChannels}")
+
+    assert(poolImage.metadata.xDim == image.metadata.xDim - convSize + 1, "Convolved image should have the right xDims.")
+    assert(poolImage.metadata.yDim == image.metadata.yDim - convSize + 1, "Convolved image should have the right yDims.")
+    assert(poolImage.metadata.numChannels == convBank.rows, "Convolved image should have the right num channels.")
+    assert(poolImage.equals(convolvedImg), "Convolved image should match theano convolution")
+  }
+
+  test("5x5 patches convolutions with stride = 2") {
+    val imgWidth = 256
+    val imgHeight = 256
+    val imgChannels = 3
+    val numFilters = 2
+    val convSize = 5
+
+    val imageBGR  = TestUtils.loadTestImage("images/test.jpeg")
+
+    val imArray = imageBGR.toArray
+
+    val imArrayRGB  = TestUtils.BGRtoRGB(new DenseMatrix(3, 256*256, imArray)).data
+
+    val image = new ChannelMajorArrayVectorizedImage(imArrayRGB, imageBGR.metadata)
+
+    val theanoFile = new File(TestUtils.getTestResourceFileName("images/theano_conv2d_2stride.csv"))
+
+    val convolvedImgRaw:Array[Double] = csvread(theanoFile).data
+
+    val convolvedImg = new ColumnMajorArrayVectorizedImage(convolvedImgRaw, ImageMetadata((imgWidth - convSize + 1)/2, (imgHeight - convSize + 1)/2, numFilters))
+
+    val convBank = convert(new DenseMatrix(2, 5*5*3, (0 until 2*5*5*3).toArray.map(x => 1)), Double)
+
+    val convolver = new Convolver(convBank, imgWidth, imgHeight, imgChannels, patchStride=2)
+
+    val poolImage = convolver(image)
+
+    logInfo(s"Image Dimensions ${poolImage.metadata.xDim} ${poolImage.metadata.yDim} ${poolImage.metadata.numChannels}")
+
+    assert(poolImage.metadata.xDim == (image.metadata.xDim - convSize + 1)/2, "Convolved image should have the right xDims.")
+    assert(poolImage.metadata.yDim == (image.metadata.yDim - convSize + 1)/2, "Convolved image should have the right yDims.")
+    assert(poolImage.metadata.numChannels == convBank.rows, "Convolved image should have the right num channels.")
+    assert(poolImage.equals(convolvedImg), "Convolved image should match theano convolution")
```

Comment:
These are nice tests! can you either put in a note or a script about how you would reproduce these CSVs in Theano?